internal/app/connectconformance/results.go

```diff
@@ -379,7 +377,9 @@ func checkPayloads(expected, actual []*conformancev1.ConformancePayload) multiEr
 			errs = append(errs, fmt.Errorf("response #%d: expecting data %x, got %x", i+1, expectedPayload.Data, actualPayload.Data))
 		}
 
-		errs = append(errs, checkRequestInfo(expectedPayload.GetRequestInfo(), actualPayload.GetRequestInfo(), i)...)
+		if i == 0 { //nolint:nestif
```

Comment:
I just realized that my suggestion changed the functionality here. It's only _part_ of `checkRequestInfo` that we skip when i != 0 -- the headers, timeout, and query params.

For that, I think `checkRequestInfo` should have a `checkHeaders bool` param. This call site can pass `i == 0` for that param, and the other call site can always provide `true`.